include/boost/unordered/detail/foa.hpp

```diff
@@ -1559,6 +1601,9 @@ table:empty_value<Hash,0>,empty_value<Pred,1>,empty_value<Allocator,2>
 #else
         std::is_trivially_copy_constructible<value_type>::value
 #endif
+        &&(
+          is_std_allocator<Allocator>::value||
```

Comment:
What about users who specialize `std::allocator`?

Reply:
Well, I guess they shouldn't :-) FWIW, stdlib implementors also assume that `std::allocator` does not have a fancy `construct` regardless of specializations:

https://github.com/llvm-mirror/libcxx/blob/master/include/memory#L1670-L1701

The way I see it, `std::memcpy` is too good not to use for the sake of corner cases such as you mention, which are in any case better served by proper user-defined allocators.

Reply:
libstdc++ also assumes `std::allocator` is not to be user specialized:
https://github.com/gcc-mirror/gcc/blob/master/libstdc%2B%2B-v3/include/bits/stl_uninitialized.h#L334-L375